core_contracts/dex/dex.py

```diff
@@ -637,6 +637,60 @@ def tokenFallback(self, _from: Address, _value: int, _data: bytes):
         else:
             revert(f"{TAG}: Fallback directly not allowed.")
 
+
+    @dex_on
+    @external
+    def transfer(self, _to: Address, _value: int, _id: int, _data: bytes = None):
+        """
+        Used to transfer LP tokens from sender to another address.
+        Calls the internal method `_transfer()` with a default
+        `_data` if none is submitted.
+        :param _to: Address to transfer to
+        :param _value: Amount of units to transfer
+        :param _id: Pool ID of token to transfer
+        :param _data: data to include with transfer
+        """
+
+        if _data is None:
+            _data = b'None'
+        self._transfer(self.msg.sender, _to, _value, _id, _data)
+
+    def _transfer(self, _from: Address, _to: Address, _value: int, _id: int, _data: bytes):
+        """
+        Used to transfer LP IRC-31 tokens from one address to another.
+        Invoked by `transfer(...)`.
+        """
+        if _value < 0:
+            revert(f"{TAG}: Transferring value cannot be less than 0.")
+        
+        if self._balance[_id][_from] < _value:
+            revert(f"{TAG}: Out of balance.")
+        
+        if _id < 5:
+            revert(f"{TAG}: untransferrable token id")
+
+        self._active_addresses[_id].add(_to)
+        if self._balance[_id][_from] == 0:
```

Comment:
Looks like this removal of the _from address from the _active_addresses set should be done after line 676 where the transfer happens or it will never fire.

Reply:
I made this change so that it can be deployed and tested on testnet. Pretty sure it is correct, but please confirm.